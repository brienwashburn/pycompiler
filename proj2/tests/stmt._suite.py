def _1(): pass
def _2():
    def _2a():
        def _2b():
            def _2c():
                def _2e():
                    try:
                        a = b
                    except Foo:
                        a = 1
                    return a
                return _2e(a)
            return _2c(a)
        return _2b(a)
    return _2b(a)
