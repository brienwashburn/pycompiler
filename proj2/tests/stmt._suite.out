(Module
 (FunctionDef
  (name _1)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body (Pass))
  (decorator_list)
  (returns #f))
 (FunctionDef
  (name _2)
  (args
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults)))
  (body
   (FunctionDef
    (name _2a)
    (args
     (Arguments
      (args)
      (arg-types)
      (vararg #f)
      (kwonlyargs)
      (kwonlyarg-types)
      (kw_defaults)
      (kwarg #f)
      (defaults)))
    (body
     (FunctionDef
      (name _2b)
      (args
       (Arguments
        (args)
        (arg-types)
        (vararg #f)
        (kwonlyargs)
        (kwonlyarg-types)
        (kw_defaults)
        (kwarg #f)
        (defaults)))
      (body
       (FunctionDef
        (name _2c)
        (args
         (Arguments
          (args)
          (arg-types)
          (vararg #f)
          (kwonlyargs)
          (kwonlyarg-types)
          (kw_defaults)
          (kwarg #f)
          (defaults)))
        (body
         (FunctionDef
          (name _2e)
          (args
           (Arguments
            (args)
            (arg-types)
            (vararg #f)
            (kwonlyargs)
            (kwonlyarg-types)
            (kw_defaults)
            (kwarg #f)
            (defaults)))
          (body
           (Try
            (body (Assign (targets (Name a)) (value (Name b))))
            (handlers
             (except
              (Name Foo)
              #f
              (Assign (targets (Name a)) (value (Num 1)))))
            (orelse)
            (finalbody))
           (Return (Name a)))
          (decorator_list)
          (returns #f))
         (Return
          (Call
           (func (Name _2e))
           (args (Name a))
           (keywords)
           (starargs #f)
           (kwargs #f))))
        (decorator_list)
        (returns #f))
       (Return
        (Call
         (func (Name _2c))
         (args (Name a))
         (keywords)
         (starargs #f)
         (kwargs #f))))
      (decorator_list)
      (returns #f))
     (Return
      (Call
       (func (Name _2b))
       (args (Name a))
       (keywords)
       (starargs #f)
       (kwargs #f))))
    (decorator_list)
    (returns #f))
   (Return
    (Call
     (func (Name _2b))
     (args (Name a))
     (keywords)
     (starargs #f)
     (kwargs #f))))
  (decorator_list)
  (returns #f)))
